### PR TITLE
Return success for core transition request from rx state to tx state

### PR DESCRIPTION
**Pull request by hubertmis**
This PR also changes transition from any state to sleep state to
return success even if there is not enough time to perfrom transition
due to ending timeslot.

### CODE REVIEW

**Comment by hubertmis** [comment]
Regression test

File: src/nrf_802154_core.c [hunk 1]
```diff
@@ -2705,18 +2726,25 @@ bool nrf_802154_core_receive(nrf_802154_term_t              term_lvl,
...
+    bool result = nrf_802154_critical_section_enter();
 
     if (result)
     {
         if ((m_state != RADIO_STATE_RX) && (m_state != RADIO_STATE_TX_ACK))
         {
-            result = current_operation_terminate(term_lvl, req_orig, notify_abort);
+            if (critical_section_can_be_processed_now())
+            {
+                result = current_operation_terminate(term_lvl, req_orig, notify_abort);
```
Comment (LuDuda) [inline src/nrf_802154_core.c]: When this can fail?

Comment (hubertmis) [reply src/nrf_802154_core.c]: Termination of current operation? Most likely when termination level is too low.

**Comment by hubertmis** [comment]
Regression test

**Pull request merged**